### ISSUE TITLE
Question:I have code question about path attribes and bgpls attribute

### ISSUE THREAD

**Issue by zhanggongc** [osrg/gobgp#2273]
NO 1:
func GetNativePathAttributes(p *api.Path) ([]bgp.PathAttributeInterface, error) {
	pattrsLen := len(p.PattrsBinary)
	if pattrsLen > 0 {
		pattrs := make([]bgp.PathAttributeInterface, 0, pattrsLen)
		for _, attr := range p.PattrsBinary {
			a, err := bgp.GetPathAttribute(attr)
			if err != nil {
				return nil, err
			}
			err = a.DecodeFromBytes(attr)
			if err != nil {
				return nil, err
			}
			pattrs = append(pattrs, a)
		}
		return pattrs, nil
	}
	return UnmarshalPathAttributes(p.Pattrs)
}
PattrsBinary  >>>what does it use for  ? i don't understand its meaning and how to use

NO 2:
func unmarshalAttribute(an *any.Any) (bgp.PathAttributeInterface, error) {
	fmt.Println("unmarshalAttribute:<<enter print>>")
	var value ptypes.DynamicAny
	if err := ptypes.UnmarshalAny(an, &value); err != nil {
		return nil, fmt.Errorf("failed to unmarshal route distinguisher: %s", err)
	}
	switch a := value.Message.(type) {
	case *api.OriginAttribute:
        ........
        ........
}
this function has no bgpls attribute process, can i expand this bgpls branch?  like this :
    case *api.LsAttribute:
    ...........
   ..........

**Comment by fujita** [comment]
No.1 IIRC, PattrsBinary can store path attributes in binary format (on wire). mrt injection uses it for the performance (avoid api conversion).

No.2 you can improve unmarshalAttribute for bgpls attributes.

**Comment by walkerxiong** [comment]
> No.1 IIRC, PattrsBinary can store path attributes in binary format (on wire). mrt injection uses it for the performance (avoid api conversion).
> 
> No.2 you can improve unmarshalAttribute for bgpls attributes.

Does this mean that the content of the two is the same, but the form is different?

Looking forward to your reply.

Regards.
